pkg/manager/config/feature_gates_test.go

```diff
@@ -1,43 +1,39 @@
-package featuregates
+package config
 
 import (
 	"fmt"
 	"testing"
 
-	"github.com/go-logr/zapr"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"go.uber.org/zap"
-
-	"github.com/kong/kubernetes-ingress-controller/v3/pkg/manager/config"
 )
 
 func TestFeatureGates(t *testing.T) {
 	t.Log("Setting up configurations and logging for feature gates testing")
-	setupLog := zapr.NewLogger(zap.NewNop())
+	// setupLog := zapr.NewLogger(zap.NewNop())
```

Comment:
```suggestion
```

Reply:
BTW removing these logs about detecting feature gates might not be desired in the context of running KIC as a CLI. Perhaps we can ignore that as most likely, on this branch, we will end up removing the CLI part completely.

Reply:
I agree, I moved the validation/logging code to the `New(...)` constructor for the internal manager, so misconfigured feature gates (and values of ones configured properly)can be reported on a manager instance level. It will be beneficial in case of KO and configuring multiple instances where each one has different set of gates enabled